Dockerfile

```diff
@@ -1,86 +1,74 @@
 # syntax = docker/dockerfile:1
 
-# Make sure RUBY_VERSION matches the Ruby version in .tool-versions
+# Define Ruby version
 ARG RUBY_VERSION=3.3
 FROM public.ecr.aws/docker/library/ruby:$RUBY_VERSION-alpine AS base
 
-# Rails app lives here
+# Set working directory
 WORKDIR /rails
 
-# Set production environment
+# Configure environment variables
 ENV RAILS_ENV="production" \
     BUNDLE_DEPLOYMENT="1" \
     BUNDLE_PATH="/usr/local/bundle" \
     BUNDLE_WITHOUT="development test" \
-    BUNDLE_FORCE_RUBY_PLATFORM="1"
+    BUNDLE_FORCE_RUBY_PLATFORM="1" \
+    RUBY_YJIT_ENABLE="true"
 
+# Stage to download the gRPC health probe
 FROM public.ecr.aws/docker/library/alpine:latest AS binaries
 
-# Install packages needed to build binaries
-RUN apk update --no-cache && \
-    apk upgrade --no-cache && \
-    apk add --update --no-cache wget curl jq
-
-RUN [[ "$(arch)" = "aarch64" ]] && arch="arm64" || arch="$(arch)" && \
-  curl -s https://api.github.com/repos/grpc-ecosystem/grpc-health-probe/releases/latest \
-  | jq ".assets[].browser_download_url" \
-  | grep "grpc_health_probe-linux-$arch" \
-  | tr -d '"' \
-  | wget -qi - -O grpc-health-probe && \
-  chmod +x grpc-health-probe && \
-  mv grpc-health-probe /usr/local/bin
-
-# Throw-away build stage to reduce size of final image
+# Install packages to download the gRPC health probe
+RUN apk add --no-cache wget curl jq
+
+# Download and configure the gRPC health probe
+RUN ARCH=$(uname -m) && \
+    [ "$ARCH" = "x86_64" ] && ARCH="amd64" || ARCH="$ARCH" && \
+    wget -q -O /usr/local/bin/grpc-health-probe \
+      $(curl -s https://api.github.com/repos/grpc-ecosystem/grpc-health-probe/releases/latest \
+      | jq -r ".assets[] | select(.name | test(\"linux-$ARCH\")) | .browser_download_url") && \
+    chmod +x /usr/local/bin/grpc-health-probe
+
+# Build stage
 FROM base as build
 
-# Install packages needed to build gems
-RUN apk update --no-cache && \
-    apk upgrade --no-cache && \
-    apk add --update --no-cache build-base git gcompat postgresql-dev nodejs yarn
+# Install build packages and temporary dependencies
+RUN apk add --no-cache --virtual .build-deps build-base git gcompat postgresql-dev nodejs yarn
 
 # Install application gems
 COPY Gemfile Gemfile.lock ./
-
 RUN bundle install && \
     rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
-    find "${BUNDLE_PATH}" -name "*.o" -delete && find "${BUNDLE_PATH}" -name "*.c" -delete && \
-    mkdir -p tmp/pids && \
```

Comment:
I think these `mkdir -p` are here because it will fail later in the build if those directories don't exist

Reply:
This item may be rolled back too.